sencha-workspace/SlateTasksStudent/app/view/TaskTree.js

```diff
@@ -192,6 +192,15 @@ Ext.define('SlateTasksStudent.view.TaskTree', {
 
                 return statusStrings[taskStatus] || '';
             },
+            getStatusDate: function(task) {
+                var taskStatus = task.TaskStatus;
+
+                if (taskStatus === 'submitted' || taskStatus === 're-submitted') {
+                    return Ext.Date.dateFormat(task.Submitted,"M d, Y");
+                } else {
+                    return Ext.Date.dateFormat(task.DueDate,"M d, Y");
+                }
```

Comment:
@BillClinton it's bad form to do an `else` after an `if` block that ends in a return.

Also this could be DRYd up a lot, use single quotes, and have spaces after function param commas:

```javascript
getStatusDate: function(taskData) {
    var taskStatus = taskData.TaskStatus,
        taskDate;

    if (taskStatus == 'submitted' || taskStatus == 're-submitted') {
        taskDate = taskData.Submitted;
    } else {
        taskDate = task.DueDate;
    }

    return Ext.Date.dateFormat(taskDate, 'M d, Y');
}
```